### ISSUE TITLE
Godot 3.2.4RC3 - M1 Mac - does not compile in Xcode

### ISSUE THREAD

**Issue by MarcusDobler** [godotengine/godot#46406]
**Godot version:**

Godot 3.2.4RC3

**OS/device including version:**

Mac OS 11.2.1, M1 Mac Book Pro 13" 

**Issue description:**

The exported project does not compile. Xcode is just stuck. Waited 15 minutes nothing happens (several times)

Expectation is running the project on my device just as before.

**Steps to reproduce:**

Export to iOS, then try to compile in Xcode 

**Minimal reproduction project:**

Works (not)
[SmallDemo.zip](https://github.com/godotengine/godot/files/6040954/SmallDemo.zip)
 with every project, I've attached a small one.

**Comment by etables** [comment]
SmallDemo works for me running on macOS Big Sur 11.1 M1 Mac Air, exported to iPhone 8 (iOS 14.4). Make sure you follow https://www.youtube.com/watch?v=BovBnO07h80, I had to change a few things in the export settings, such as specifying a name for the app, and removing spaces from the .ipa file.

**Comment by akien-mga** [comment]
You likely have to wait more than 15 min, the iOS binaries in 3.2.4 RC 2 and RC 3 are compiled with Link Time Optimization, which means that optimization happens when linking, and this is a fairly slow process (but produces better optimized binaries, thus better performance).

Still, we'll go back to disabling LTO for RC 4 as it seems to be too slow for most users to accept.

Duplicate of #46064.

**Comment by MarcusDobler** [comment]
Ok, I used Xcode to debug on device If I've have to wait 20 minutes for every compilation its indeed a little slow.

**Comment by ViktorEvil** [comment]
It only takes that long on the first compile as you can just export the .pck file in Godot and drop it into your Xcode project  after that.

**Comment by MarcusDobler** [comment]
Thank you, I do not use the .pck I embed the GDScript Code Folder in Xcode so changes in my GDScript code will take effect without reexporting every time I change! This is what I meant with I use Xcode to debug on my device. Does this also only take one time so long?

**Comment by ViktorEvil** [comment]
> Thank you, I do not use the .pck I embed the GDScript Code Folder in Xcode so changes in my GDScript code will take effect without reexporting every time I change! This is what I meant with I use Xcode to debug on my device. Does this also only take one time so long?

Yes only once

**Comment by naithar** [comment]
@MarcusDobler you can also use unofficial binaries that I gave a link to in #46064. 
You'll have to replace `.a` file with the new one. This should simplify testing and speed up rebuilds if they do happen.

**Comment by MarcusDobler** [comment]
I do not want to annoy you. This is running for 30 Minutes now. Usually this M1 Mac ist pretty fast! And worst thing its showing that it is idle. 
<img width="1247" alt="image" src="https://user-images.githubusercontent.com/16276195/109175254-ea49e880-7785-11eb-96be-c172fe03d44b.png">

**Comment by MarcusDobler** [comment]
WOW sorry in this minute it finished! Its running now. Sorry! But why the compilation process does not use more CPU Power? Anyway Sorry!

**Comment by naithar** [comment]
I’m not sure about CPU, but I guess more RAM the better - at least on my M1 macbook with 16GB RAM it’s taking 5-10 minutes to finish linking with lto binaries.
Just use unofficial non-lto binaries for now (there are also ios templates you can use in case you want to perform export multiple times). 
Next RC should fix this.

**Comment by Calinou** [comment]
LTO is very demanding in terms of RAM. It's recommended to have 16 GB of RAM on your system or more.

**Comment by generic-glitch** [comment]
This seems to be closed but on the most recent version of iOS I have waited for over 20 min with a M1 MacBook Pro (16gb ram 512gb ssd) and nothing happens, always just stuck on archive. Not sure what to do about it.

**Comment by Calinou** [comment]
> This seems to be closed but on the most recent version of iOS I have waited for over 20 min with a M1 MacBook Pro (16gb ram 512gb ssd) and nothing happens, always just stuck on archive. Not sure what to do about it.

Which Godot version did you compile, and which SCons options did you specify?

**Comment by generic-glitch** [comment]
> > This seems to be closed but on the most recent version of iOS I have waited for over 20 min with a M1 MacBook Pro (16gb ram 512gb ssd) and nothing happens, always just stuck on archive. Not sure what to do about it.
> 
> Which Godot version did you compile, and which SCons options did you specify?

Oh my bad, I didn't use any SCons options. I just used the release from the godotengine website. Nevermind, after 10 minutes it compiled anyways and it didnt happen again after so its good now.
